Gems/AtomTressFX/Assets/Passes/HairFillPPLL.pass

```diff
@@ -0,0 +1,130 @@
+{
+    "Type": "JsonSerialization",
+    "Version": 1,
+    "ClassName": "PassAsset",
+    "ClassData": {
+        "PassTemplate": {
+            "Name": "HairPPLLRasterPassTemplate",
+            "PassClass": "HairPPLLRasterPass",
+            "Slots": [
+                // Input/Outputs...
+                {
+                    // Super important to keep this as it is used to set the size of the Head PPLL image buffer.
+                    // If DepthLinear is not availbale - connect to another viewport (non MSAA) image.
```

Comment:
Indeed - a bit of explanation is required unless you dive deeper.
I need a texture that is passed as texture and not RT in order to declare as texture with sampler and get dimensions - if the texture is used as an RT in the shader, it cannot be also requested for sampling (hence depth and color cannot be considered).
The reason for non-MSAA is different as this shader is on purpose done in non-MSAA mode to save on extremely high performance loss.   The use of MSAA to down sample is done with very light pass before the raster pass and it copies it before from MSAA to non-MSAA RT to be used here.
So.. two separate topics but it's important to mention both with assumption that a user that will want to change already did a deep dive and understands what is done and this is only a reminder.
|Let me know if I can clear the topic

Reply:
If it is only used in the shader to get dimensions, I would suggest just passing the texture width/height and inverse width/height as a float4 in the SRG. This will save you from having to bind an extra texture and will also be faster in the shader.

Reply:
I agree this would be the best method.  I asked Sid, Vicky and you and tried to look in the code for that (for example, I expected it to be found already in the PerView Srg), but the truth is that Atom is not providing this and for a reason.  All of the frame buffers are also kept transient and so all the shaders in Atom are using my method.
In practice it's not a bad method since you are dependent on the size of RT coming to you and this can be changed in various ways and you don't mind as you'd get it.  As I said, I would have liked like you to also establish it in the PerView Srg for ease of use and redundancy but me and other people could not find an easy way that will not expose some bugs (I tried from the camera / scene / etc.. there is not good error safe consensus).
Let me know if this satisfices your question - this can always be changed easily if / when Atom exposes this.